contracts/Forwarder.sol

```diff
@@ -36,40 +38,114 @@ contract Forwarder is AccessControlEnumerable, Pausable {
+    }
+
+    /// @notice Set validate selector
+    /// @dev Allows admin to set validate selector
+    /// @param _validateSelector validate selector
+    function setValidateSelector(bytes4 _validateSelector)
+        external
+        onlyRole(FORWARDER_ADMIN_ROLE)
+    {
+        validateSelector = _validateSelector;
+    }
+
+    /// @notice pause the contract
     function pause() external onlyRole(PAUSE_ROLE) {
         Pausable._pause();
     }
 
+    /// @notice unpause the contract
     function unpause() external onlyRole(PAUSE_ROLE) {
         Pausable._unpause();
     }
 
-    /// @notice get result by collection name
-    function getResult(bytes32 collectionName)
+    /**
+     * @notice Updates the result based on the provided data and returns the latest result
+     * @param data bytes data required to update the result
+     * @return result of the collection, its power and timestamp
+     */
+    function getResult(bytes calldata data)
+        external
+        whenNotPaused
+        onlyRole(TRANSPARENT_FORWARDER_ROLE)
+        returns (uint256, int8, uint256)
+    {
+        
+        require(
+            updateSelector != bytes4(0),
+            "No update selector"
+        );
+        bytes memory returnData = resultManager.functionCall(
+            abi.encodePacked(
+                updateSelector,
+                data
+            )
+        );
+        return abi.decode(returnData, (uint256, int8, uint256));
+    }
+
+    /**
+     * @dev using the hash of collection name, clients can query the result of that collection
+     * @param name bytes32 hash of the collection name
+     * @return result of the collection and its power
+     */
+    function getResult(bytes32 name)
         external
         view
         whenNotPaused
         onlyRole(TRANSPARENT_FORWARDER_ROLE)
-        returns (uint256, int8)
+        returns (uint256, int8, uint256)
     {
         require(
-            collectionPayload[collectionName].length > 0,
-            "Invalid collection name"
+            resultGetterSelector != bytes4(0),
+            "No result getter selector"
+        );
+        bytes memory returnData = resultManager.functionStaticCall(
+            abi.encodePacked(
+                resultGetterSelector,
+                name
+            )
         );
-        bytes memory data = resultManager.functionStaticCall(
-            collectionPayload[collectionName]
+        return abi.decode(returnData, (uint256, int8, uint256));
+    }
+
+    /**
+     * @dev validates the result based on the provided data and returns the validity
+     * @param data bytes data required to validate the result
+     * @return validity of the result
+     */
+    function validateResult(bytes calldata data)
+        external
+        view
+        whenNotPaused
+        onlyRole(TRANSPARENT_FORWARDER_ROLE)
+        returns (bool)
+    {
+        require(
+            validateSelector != bytes4(0),
```

Comment:
we can add a modifier for this check, and maybe we can migrate to error types. Use modifier wherever required.

Reply:
https://github.com/razor-network/bridge-proxy/pull/59/commits/00222f0d6f0d9912d66935ac1fd7dde93da0f06b

added modifier

Reply:
https://github.com/razor-network/bridge-proxy/pull/59/commits/857e9caac099e33c24e6fc54bb698f1e385f2821
Added custom error types